pkg/policy/condition/key.go

```diff
@@ -106,6 +118,25 @@ var CommonKeys = []Key{
 	AWSSecureTransport,
 	AWSCurrentTime,
 	AWSEpochTime,
+	AWSPrincipalType,
+	AWSUserID,
+	AWSUsername,
+}
+
+var varStr = func(k Key) string {
+	return fmt.Sprintf("${%s}", k)
+}
+
+// CommonVariables - is list of all common policy variables for substitution.
+var CommonVariables = []string{
+	varStr(AWSSourceIP),
+	varStr(AWSUserAgent),
+	varStr(AWSSecureTransport),
+	varStr(AWSCurrentTime),
+	varStr(AWSEpochTime),
+	varStr(AWSPrincipalType),
+	varStr(AWSUserID),
+	varStr(AWSUsername),
```

Comment:
Is it good idea to use `set.StringSet`?  I think `init()` is better place to compute like `varStr()` inside.

Reply:
Let's see

Reply:
I see `CommonKeys` and `CommonVariables` are related by index.  This will break (currently as well) if some reorder happening in add/remove.

I think its better to add a method `func (key Key) VarName() string` which will return respective variable names.  This way below code is simpler.
```go
	fvalues := f.values.ApplyFunc(func(v string) string {
	fvalues := f.values.ApplyFunc(func(v string) string {
		for _, key := range CommonKeys {
			if rvalues, ok := values[key.VarName()]; ok && rvalues[0] != "" {
				v = strings.Replace(v, key.VarName(), rvalues[0], -1)
			}
		}

		return v
	})
```

Reply:
this makes sense..